src/editors/containers/ProblemEditor/data/ReactStateOLXParser.js

```diff
@@ -0,0 +1,253 @@
+    const demandhint = this.addHints();
+    const problemObject = {
+      problem: {
+        [problemType]: {
+          ...question,
+          [widget]: widgetObject,
+        },
+        ...demandhint,
+      },
+    };
+    return this.builder.build(problemObject);
+  }
+
+  buildTextInput() {
+    const question = this.addQuestion();
+    const demandhint = this.addHints();
+    const answerObject = this.buildTextInputAnswersFeedback();
+    const problemObject = {
+      problem: {
+        [OLXProblemTypes.TEXTINPUT]: {
+          ...question,
+          ...answerObject,
+        },
+        ...demandhint,
+      },
+    };
+    return this.builder.build(problemObject);
+  }
+
+  buildTextInputAnswersFeedback() {
+    const { answers } = this.problemState;
+    let answerObject = {};
+    const additionAnswers = [];
+    const wrongAnswers = [];
+    let firstCorrectAnswer = false;
+    answers.forEach((answer) => {
+      const correcthint = this.getAnswerHints(answer);
+      if (answer.correct && firstCorrectAnswer) {
+        additionAnswers.push({
+          '@_answer': answer.title,
+          ...correcthint,
+        });
+      } else if (answer.correct && !firstCorrectAnswer) {
+        firstCorrectAnswer = true;
+        answerObject = {
+          '@_answer': answer.title,
+          ...correcthint,
+        };
+      } else if (!answer.correct) {
+        wrongAnswers.push({
+          '@_answer': answer.title,
+          '#text': answer.feedback,
+        });
+      }
+    });
+    answerObject = {
+      ...answerObject,
+      additional_answer: additionAnswers,
+      stringequalhint: wrongAnswers,
+      '@_type': _.get(this.problemState, 'additionalAttributes.type', 'ci'),
+      textline: {
+        '@_size': _.get(this.problemState, 'additionalAttributes.textline.size', 20),
+      },
+    };
+    return answerObject;
+  }
+
+  buildNumericInput() {
+    const question = this.addQuestion();
+    const demandhint = this.addHints();
+    const answers = this.buildNumericalResponse();
+    const problemObject = {
+      problem: {
+        ...question,
+        [OLXProblemTypes.NUMERIC]: answers,
+        ...demandhint,
+      },
+    };
+    return this.builder.build(problemObject);
+  }
+
+  buildNumericalResponse() {
+    const { answers } = this.problemState;
+    const answerArray = [];
+    answers.forEach((answer) => {
+      if (answer.correct) {
+        let responseParam = {};
+        const correcthint = this.getAnswerHints(answer);
+        if (_.has(answer, 'tolerance')) {
+          responseParam = {
+            responseparam: {
+              '@_type': 'tolerance',
+              '@_default': _.get(answer, 'tolerance', 0),
+            },
+          };
+        }
+        answerArray.push({
+          formulaequationinput: {
+            '#text': '',
+          },
+          '@_answer': answer.title,
+          ...responseParam,
+          ...correcthint,
+        });
+      }
+    });
```

Comment:
This is problemetic when multiple correct answers are configured. In this case, the `<additional_answer>` tag is expected with the extra answer inside a single `<numericalresponse>` tag.

However, we are returing multiple `<numericalresponse>` tags, which results in the studio UI rendering multiple input boxes :

![image](https://user-images.githubusercontent.com/36229857/196182099-64b089a9-aa53-4735-95df-72d6481a9f58.png)

Reply:
Good catch! for now I have converted these to additional answers but I do have a discussion with Conor going on.